tests/test_config.py

```
@@ -1,10 +1,302 @@
+        """Test loading only YAML configuration successfully with all critical keys."""
+        mock_exists.side_effect = lambda path: path == 'config.yaml' # Only config.yaml exists
+        
+        # Sample critical keys that would be in YAML
+        # For this test, we ensure all CRITICAL_CONFIG_KEYS are present and valid
+        # to pass validate_critical_config.
+        yaml_data = {
+            'newspaper': {'url': 'https://example.com', 'username': 'user', 'password': 'pass'},
+            'email': {'recipient': 'r@example.com', 'sender': 's@example.com', 
+                      'smtp_server': 'smtp.example.com', 'smtp_port': 587, 
+                      'smtp_username': 'smtp_user', 'smtp_password': 'smtp_pass'},
+            'paths': {'download_dir': '/tmp/downloads'}
```

Ruff (0.11.9)

Probable insecure usage of temporary file or directory: "/tmp/downloads"

(S108)